api/src/domain/user/searchAndAddTimeStampToBusinessSearch.ts

```diff
@@ -0,0 +1,33 @@
+import { NameAvailabilityResponse, NameAvailabilityStatus } from "@shared/businessNameSearch";
+import { getCurrentDateISOString } from "@shared/dateHelpers";
+import { BusinessNameClient } from "../types";
+
+export const searchAndAddTimeStampToBusinessSearch = async (
+  businessName: string,
+  businessNameClient: BusinessNameClient
+): Promise<BusinessNameSearchResponse> => {
```

Comment:
I would encourage you to follow the pattern we have for other domain functions that they are a function that returns a function.

The function signature here is a little strange to use because it requires a business name (something that changes on every single query) and a businessNameClient (something that is an object that is always the same).

The reason we have a function-that-returns-a-function is allow the top-level (the app.ts file) create an inject all the necessary dependencies. Here, we're requiring the caller of this function (the routers) to inject the dependency instead of just setting them up ahead of time.

This will simplify the tests you wrote for the routers, which are right now intrinsically testing the behavior of this function as well. Instead, you would pass an instance of a searchAndAddTimeStampToBusinessSearch function as a dependency to the router (instead of a business name client). Think of it this way. The router is never calling a function on the businessNameClient - and it never should. it should ALWAYS use this searchAndAddTimeStampToBusinessSearch to search a name. So why are we giving it access to the whole client? (fun fact - in SOLID principles, this this the "I")